calico/common.py

```diff
@@ -71,18 +75,27 @@
     "protocol",
     "src_net",
     "src_tag",
+    "src_selector",
     "src_ports",
     "dst_net",
     "dst_tag",
+    "dst_selector",
     "dst_ports",
     "icmp_type",
     "icmp_code",
     "ip_version",
 ])
 
+# Valid actions to see in a rule.
+KNOWN_ACTIONS = set(["allow", "deny", "next-tier"])
+
 # Regex that matches only names with valid characters in them. The list of
 # valid characters is the same for endpoints, profiles, and tags.
-VALID_ID_RE = re.compile('^[a-zA-Z0-9_\.\-]+$')
+VALID_ID_RE = re.compile(r'^[a-zA-Z0-9_\.\-]+$')
+
+# Regex for validating the names of labels, which need to be rich enough to
+# allow for Kubernetes implementation, for example.
+VALID_LABEL_NAME_RE = re.compile(r'^[%s]+$' % re.escape(LABEL_CHARS))
```

Comment:
Is that the right kind of escaping?  Different escaping is needed within [ ] than when specifying a literal string.  (For example, - has a special meaning within [ ], so would need escaping, whereas it doesn't as a literal.)